shpc/main/modules/views.py

```diff
@@ -0,0 +1,504 @@
+        """
+        init of a view must be done when it exists, and from the base.ModuleBase
+        (instantiated with either lmod or tcl). There is no concept of a view
+        without this central installation.
+        """
+        self.name = name
+        self.settings = settings
+        self.symlink_extension = symlink_extension
+        self.module_extension = module_extension
+        self.versionfile = versions.VersionFile(self.settings, self.module_extension)
+        self.modulefile = modulefile
+        self.reload()
+
+    @property
+    def path(self):
+        """
+        The path is guaranteed to exist because the View is not instantiated
+        if it does not.
+        """
+        return os.path.join(self.settings.views_base, self.name)
+
+    def reload(self):
+        """
+        Reload the view's config (given a change)
+        """
+        self._config = utils.read_yaml(self.config_path)
+
+    def symlink_exists(self, module_dir):
+        """
+        Return True or false if the view to a specific version exists or not.
+        This check is looking for a symlink.
+        """
+        return os.path.islink(self.get_symlink_path(module_dir))
+
+    def exists(self, module_dir):
+        """
+        Determine if a specific version is symlinked OR there is a matching
+        directory for a module. This exists is intended for looking to see
+        if a module_dir pattern (either to a directory or version) exists.
+        E.g.,:
+
+        shpc uninstall ghcr.io/autamus/clingo:5.5.1 will link to a lua file
+        shpc uninstall ghcr.io/autamus/clingo will point to a directory.
+        """
+        symlink_path = self.get_symlink_dir(module_dir, has_version=False)
+        return os.path.exists(symlink_path) or self.symlink_exists(module_dir)
+
+    def _sub_module_base(self, path):
+        return path.replace(self.settings.module_base, "$module_base")
+
+    def _sub_views_base(self, path):
+        return path.replace(self.settings.views_base, "$views_base")
+
+    @property
+    def config_path(self):
+        """
+        Get the config path to the view
+        """
+        return os.path.join(self.path, "view.yaml")
+
+    def confirm_install(self, module_dir, force=False):
+        """
+        Check if a module install exists under a view. A module install to a view
+        is a symbolic link, so here we are checking for existence of that. If
+        force is True, we continue even if it already exists.
+        """
+        if self.symlink_exists(module_dir):
+            if force:
+                logger.info(
+                    "Overwriting view %s install of %s, as requested"
+                    % (self.name, module_dir)
+                )
+            elif not utils.confirm_action(
+                "%s in view %s already exists, are you sure you want to overwrite"
+                % (self._sub_module_base(module_dir), self.name)
+            ):
+                sys.exit(0)
```

Comment:
Hi @vsoch . When re-installing a module that's already been installed, `shpc` asks the user to confirm whether they want to overwrite the symlink (code highlighted here), but not whether they want to replace the actual module itself. Don't you think it should first (and perhaps only) ask about the module ?

Reply:
The assumption is that re-installing is fairly harmless, it would regenerate "the same" file. A symlink, on the other hand, could be a totally different module with the same name.